app/android/app_template/src/org/xwalk/app/template/AppTemplateActivity.java

```diff
@@ -35,24 +48,35 @@ protected void didTryLoadRuntimeView(View runtimeView) {
     }
 
     private void enterFullscreen() {
-        if (VERSION.SDK_INT >= VERSION_CODES.KITKAT &&
-                ((getWindow().getAttributes().flags &
-                        WindowManager.LayoutParams.FLAG_FULLSCREEN) != 0)) {
+        if (isNewerThanKitkat() && ((getWindow().getAttributes().flags &
+                WindowManager.LayoutParams.FLAG_FULLSCREEN) != 0)) {
             View decorView = getWindow().getDecorView();
-            decorView.setSystemUiVisibility(
-                    View.SYSTEM_UI_FLAG_LAYOUT_STABLE |
-                    View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION |
-                    View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN |
-                    View.SYSTEM_UI_FLAG_HIDE_NAVIGATION |
-                    View.SYSTEM_UI_FLAG_FULLSCREEN |
-                    View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
+            decorView.setSystemUiVisibility(SYSTEM_UI_OPTIONS);
         }
     }
 
     public void setIsFullscreen(boolean isFullscreen) {
         if (isFullscreen) {
             enterFullscreen();
+            setSystemUiVisibilityChangeListener();
         }
     }
 
+    private void setSystemUiVisibilityChangeListener() {
+        if (isNewerThanKitkat()) {
```

Comment:
setSystemUiVisibilityChangeListener should be a pure set method. It's better to remove the version check outside.

```
if (isNewerThanKitkat())
    setSystemUiVisibilityChangeListener()
```